execution/executors/shim/process.go

```diff
@@ -123,6 +130,7 @@ func loadProcess(root, id string) (*process, error) {
 	}()
 
 	p := &process{
+		// ctx is not set here (should we set context.TODO?)
```

Comment:
The context is set in the caller on success at the moment. since your PR change `newProcess` to receive it via the opts struct, maybe just pass it as an argument here.